js/ui.js

```diff
@@ -31,12 +31,50 @@ function toggle_selected(item,extra) {
   if(selected.length > 0 || extraselected.length > 0) {
     var csel = get_class_from_classlist(selected.concat(extraselected)[0]);
     var cdsel = selected.concat(extraselected)[0].closest("div");
-    // Select only things of the same type for now - editing
-    // relations to add things means deleting and re-adding
-    if(ci != csel)
-      return;
     if(cd != cdsel)
       return;
+    // The attempted selection is a note on top of a single already selected relation.
+    if(ci != csel) {
+      if (csel == "relation" && selected.length + extraselected.length == 1 && ci == "note") {
+        // Obtain the relevant relation and note id's from the SVG.
+        var note_id = item.id;
+        var isExtraSelected = (extraselected.length == 1);
+        var relation_id = selected.concat(extraselected)[0].getAttribute("id");
+        // Search for the MEI arc linking said relation and note.
+        var arcs = Array.from(mei_graph.getElementsByTagName("arc"))
+                  .filter((x) => {
+                    return (x.getAttribute("to") == ("#gn-" + note_id) &&
+                            x.getAttribute("from") == ("#" + relation_id));
+                  })
+        // If the arc is found (as it should), edit the arc.
+        if (arcs.length == 1) {
+          var arc = arcs[0];
+          var mei_relation = Array
+                  .from(mei_graph.getElementsByTagName("node"))
+                  .filter((x) => x.getAttribute("xml:id") == relation_id)
+                  [0];
+          var new_type = extra ? "primary" : "secondary";
+          arc.setAttribute("type", new_type);
+          // Delete and redraw the hull.
+          selected.concat(extraselected)[0].parentElement.removeChild(selected.concat(extraselected)[0]);
+          selected.shift();
+          extraselected.shift();
+          redrawn = draw_relation(current_draw_context, mei_graph, mei_relation)[0];
```

Comment:
This should properly be done for all the instances of the relation, no matter the draw context (or if it is in the hierarchy visualisation).

It might conceivably be easier to just do something similar to what's done when the width slider changes value (or even abstract that whole process into something like `redraw_relation(draw_context, mei_relation)`), but for just the single relation.

Reply:
I'll sit down and see what I can do.

Reply:
Indeed, @pettter. Clearly an oversight on my part. If this costs you too much time I could work on it tomorrow evening. Thanks.